### PR TITLE
 Added shuttle arrival timer to PDA

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Just see screenshots.

## Why / Balance

Because I want.

## Technical details

Idk.

## Media

image
image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: VigersRay
- add: Added shuttle timer to PDA.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
finally, no more "[Common] Urist Mchands: "eeerrrmmmmm shuttle eta?""

**Comment by Dragonjspider** [comment]
Maybe also change the shuttle call announcement to mention to check your PDA for details? not necessary, but would be nice for newer players.

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs [hunk 1]
```diff
@@ -77,9 +80,15 @@ public override void Initialize()
...
         SubscribeNetworkEvent<EmergencyShuttleRequestPositionMessage>(OnShuttleRequestPosition);
+        SubscribeLocalEvent<RoundEndTextAppendEvent>(OnRoundEnded);
         InitializeEmergencyConsole();
     }
 
+    private void OnRoundEnded(RoundEndTextAppendEvent ev)
+    {
+        DockTime = null;
+    }
+
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs:91]: Use https://github.com/space-wizards/space-station-14/pull/19471 and device networking.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs (hunk 1 above)
Comment (graevy) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs:91]: > So i just wait merge shuttle timers and after merge get dock time from timers?

the docktime is stored in the ArrivalsShuttleComponent (NextTransfer)

Comment (VigersRay) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs:91]: > > So i just wait merge shuttle timers and after merge get dock time from timers?
> 
> the docktime is stored in the ArrivalsShuttleComponent (NextTransfer)

But i need Evacuation shutte, not arriaval

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.Console.cs [hunk 2]
```diff
@@ -33,6 +33,8 @@ public sealed partial class EmergencyShuttleSystem
     /// </summary>
     private float _consoleAccumulator = float.MinValue;
 
+    public float СonsoleAccumulator => _consoleAccumulator;
```
[the remark this answers is not in the document]
Comment (graevy) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.Console.cs:36]: oh, sorry i misread from the title. that's not stored internally but grabbed from a cvar when the shuttle is called and spawns a timer/countdown token for recalls. if we're at a point where 3 things are independently tracking the evac shuttle data then we should refactor the emergency shuttle and console system to store this in a component; the console system is jank as hell.

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs (hunk 1 above)
Comment (graevy) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs:91]: > > > So i just wait merge shuttle timers and after merge get dock time from timers?
> > 
> > 
> > the docktime is stored in the ArrivalsShuttleComponent (NextTransfer)
> 
> But i need Evacuation shutte, not arrival

oh, sorry i misread from the title. that's not stored internally but grabbed from a cvar when the shuttle is called and spawns a timer/countdown token for recalls. if we're at a point where 3 things are independently tracking the evac shuttle data then we should refactor the emergency shuttle and console system to store this in a component; the console system is jank as hell.

though this won't be as necessary once the hallway screens are merged

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.Console.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (VigersRay) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.Console.cs:38]: now used)

File: Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs (hunk 1 above)
Comment (VigersRay) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs]: > Use #19471 and device networking.

i dont understand how use networking in this case.

Comment (VigersRay) [reply Content.Server/Shuttles/Systems/EmergencyShuttleSystem.cs]: > Use #19471 and device networking.

I remove DockTime = null and use other solve. Please check this.

**Comment by VigersRay** [comment]
I don’t know how to get time from the shuttle timers, and I don’t really want to catch packets through a PDA.

**Comment by metalgearsloth** [comment]
> I don’t know how to get time from the shuttle timers, and I don’t really want to catch packets through a PDA.

It's better than hardcoding every PDA program into the system.

**Comment by VigersRay** [comment]
> > I don’t know how to get time from the shuttle timers, and I don’t really want to catch packets through a PDA.
> 
> It's better than hardcoding every PDA program into the system.

does this mean I should only receive network packets or first request them and then receive them?
And does this mean that all the data that is on the PDA should be received via network packets and not by hardcode?

**Comment by juliangiebel** [comment]
Here is the device networking docs:
https://docs.spacestation14.com/en/space-station-14/device-network.html

**Comment by TheShuEd** [comment]
I'm closing this pr as stale. If you are still working on it, please let us know and we will reopen it.

**Pull request closed without merging**